v2/bigtable-common/src/main/java/com/google/cloud/teleport/v2/bigtable/transforms/BigtableConverters.java

```diff
@@ -58,18 +73,93 @@ public Mutation apply(SchemaAndRecord record) {
+      Long columnTs = getAndVerifySchemaFields(rowkey, row, timestampColumn());
+
       for (TableFieldSchema column : columns) {
         String columnName = column.getName();
-        if (columnName.equals(rowkey())) {
+
+        if ((columnName.equals(rowkey()))
+            || (columnTs != null && columnName.equals(timestampColumn()))) {
           continue;
         }
 
         Object columnObj = row.get(columnName);
         byte[] columnValue = columnObj == null ? null : Bytes.toBytes(columnObj.toString());
         // TODO(billyjacobson): handle other types and column families
-        put.addColumn(Bytes.toBytes(columnFamily()), Bytes.toBytes(columnName), columnValue);
+
+        // If skipNullValues is true and columnValue is null, we skip adding the column.
+        // Otherwise, we proceed to add the column.
+        if (!(skipNullValues() && columnValue == null)) {
+
+          // set cell timestamp if specified and exists
+          if (columnTs != null) {
+            put.addColumn(
+                Bytes.toBytes(columnFamily()), Bytes.toBytes(columnName), columnTs, columnValue);
+          } else {
+            put.addColumn(Bytes.toBytes(columnFamily()), Bytes.toBytes(columnName), columnValue);
+          }
+        }
       }
       return put;
     }
+
+    /**
+     * If TimestampColumn is configured, retrieve timestamp value from a {@link GenericRecord} that
+     * represents the timestamp of the cell in bigtable. The value is expected to be milliseconds
+     * precision with hbase client handling to micros granularity, e.g. UNIX_MILLIS(timestamp). This
+     * method also performs basic verification checks to avoid inadvertent schema or data issues,
+     * including: 1) If the field is missing (null), it logs a warning and the default timestamp on
+     * write will be used, 2) If the field is present, it then attempts to cast it to a {@link
+     * Long}.
+     *
+     * @param rowkey The unique identifier for the row being processed. Used for clearer
+     *     error/warning messages.
+     * @param row The {@link GenericRecord} from which to extract the timestamp.
+     * @param timestampColumnName The name of the column expected to contain the timestamp value.
+     * @return The timestamp value as a {@link Long} if successfully retrieved and castable; returns
+     *     {@code null} if the timestamp column is not found in the row (a warning will be logged).
+     * @throws IllegalArgumentException If the timestamp column is found but cannot be cast to a
+     *     {@link Long}, or if an {@link AvroRuntimeException} occurs during field access.
+     */
+    private Long getAndVerifySchemaFields(
+        String rowkey, GenericRecord row, String timestampColumnName) {
+      Long columnTs = null;
+
+      // timestamp column is optional, if it's specified, retrieve the field value
+      if (timestampColumnName != null && timestampColumnName.length() > 0) {
```

Comment:
nit: 
```suggestion
      if (timestampColumnName != null && !timestampColumnName.isEmpty()) {
```

Also the default value of timestampColumnName is "" so would it ever be null?